moondance-labs/tanssi
# Implement on demand block production

Allows registered parathreads to produce blocks every n slots, where n is defined on chain.

This is ready to be tested, but it may need a refactor (`get_authorities_from_orchestrator` will probably need to be renamed)

How to test this:

* `pnpm moonwall test zombie_tanssi_parathreads`

This is a new zombienet test suite that is not being run in CI.

Or, to test manually:

* `pnpm moonwall run zombie_tanssi`

* Manually call Registrar.deregister for para id 2000

* `pnpm run sudo-register-para register --parathread --url ws://127.0.0.1:9948 --account //Alice --chain specs/template-container-2000.json`

* Manually call Registrar.setParathreadParams to adjust slotFrequency. Set min=max=5 to produce 1 block every 5 slots (60 seconds)

* Observe how parachain 2000 produces 1 block every 60 seconds.
